common/src/node.rs

```diff
@@ -95,26 +290,69 @@ impl VoxelData {
             VoxelData::Solid(mat) => mat,
         }
     }
+
+    pub fn is_solid(&self) -> bool {
+        match *self {
+            VoxelData::Dense(_) => false,
+            VoxelData::Solid(_) => true,
+        }
+    }
+
+    pub fn from_serializable(serializable: &SerializableVoxelData, dimension: u8) -> Option<Self> {
+        if serializable.voxels.len() != usize::from(dimension).pow(3) {
+            return None;
+        }
+
+        let mut data = vec![Material::Void; (usize::from(dimension) + 2).pow(3)];
+        let mut input_index = 0;
+        for x in 0..dimension {
+            for y in 0..dimension {
+                for z in 0..dimension {
+                    data[Coords([x, y, z]).to_index(dimension)] = serializable.voxels[input_index];
+                    input_index += 1;
+                }
+            }
+        }
+        Some(VoxelData::Dense(data.into_boxed_slice()))
+    }
+
+    pub fn to_serializable(&self, dimension: u8) -> SerializableVoxelData {
```

Comment:
This and `from_serializable` might benefit from a test. I accidentally left a bug in for months where this function simply didn't work (due to me mixing up vec capacity and length). It is relatively simple, though, so that might be overkill.

Reply:
Definitely worth having end-to-end serialization tests, which should cover the simpler components.

Reply:
Would it make sense for me to add them to this PR? If so, where would you recommend I begin? Should they be a test that pairs `to_serializable` and `from_serializable`, or should it be higher level than that? If it's the latter, I'm a bit lost on where to begin, as I still don't have a full grip on what should be play-tested and what should be automatically tested in various stages of development of a game.

I'm happy to add a unit test here, or add one or more higher-level tests if you are willing to help mentor me. My main questions would be (1) where should these tests live, and (2) what is an example of a thing that a particular test would test?

Alternatively, these tests might be out of scope for this PR.

Reply:
I was thinking in terms of "serialize and deserialize an entire world and check if it's unchanged". This PR doesn't introduce that feature, but we'll want it in the near future, so I'm happy to punt for now.

Reply:
Got it. I'd expect such a test to be tricky and involved enough to write (since by default, saves and loads should generally only operate on parts of the world, not the whole world), that the test itself will likely deserve its own PR.